lib/note-info/reference.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Fragment, FunctionComponent } from 'react';
+import { connect } from 'react-redux';
+import format from 'date-fns/format';
+
+import actions from '../state/actions';
+import { getNoteReference } from '../utils/get-note-references';
+
+import type * as S from '../state';
+import type * as T from '../types';
+
+type OwnProps = {
+  noteId: T.EntityId;
+};
+
+type StateProps = {
+  reference:
+    | {
```

Comment:
Why are these pipes here

Reply:
The second pipe is an or. So it is the object or undefined. Prettier added the top one so it could add a line break.